test/test_dynamictablesummary.py

```diff
@@ -0,0 +1,114 @@
+            obs_intervals=[[1, 10]],
+            location="CA1",
+            quality=0.95,
+            isi_violation=0.1,
+            firing_rate=5.1
+        )
+        self.nwbfile.add_unit(
+            spike_times=[2.2, 3.0, 25.0, 26.0],
+            obs_intervals=[[1, 10], [20, 30]],
+            location="CA3",
+            quality=0.85,
+            isi_violation=0.12,
+            firing_rate=2.1
+        )
+        self.nwbfile.add_unit(
+            spike_times=[1.2, 2.3, 3.3, 4.5],
+            obs_intervals=[[1, 10], [20, 30]],
+            location="CA1",
+            quality=0.90,
+            isi_violation=0.3,
+            firing_rate=10.1
+        )
+        self.nwbfile.add_unit(
+            spike_times=[1.4, 2.4, 3.4, 4.6],
+            obs_intervals=[[1, 10], [20, 30]],
+            location="CA3",
+            quality=0.92,
+            isi_violation=0.21,
+            firing_rate=14.4
+        )
+
+    def test_psth_widget(self):
+        widget = DynamicTableSummaryWidget(self.nwbfile.units)
+        assert isinstance(widget, widgets.Widget)
+
+        nbins = 2
+        col_names_display = widget.col_names_display.keys()
+        real_names = [col for col in col_names_display if "(r)" in col]
+        cat_names = [col for col in col_names_display if "(c)" in col]
+        
+        # select 1 real
+        cols_to_plot = (real_names[0])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
+        
+        # select 1 cat
+        cols_to_plot = (cat_names[0])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
+        
+        # select 1 real + 1 cat
+        cols_to_plot = (real_names[0], cat_names[0])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
+        
+        # select 2 real
+        cols_to_plot = (real_names[0], real_names[1])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
+        
+        # select 2 real + 1 cat
+        cols_to_plot = (real_names[0], real_names[1], cat_names[0])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
+        
+        # select 3 real
+        cols_to_plot = (real_names[0], real_names[1], real_names[2])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
+        
+
+        # select 3 real + 1 cat
+        cols_to_plot = (real_names[0], real_names[1],
+                        real_names[2], cat_names[0])
+        widget.controls["col_names_display"] = cols_to_plot
+        widget.plot_hist_bar(col_names_display=cols_to_plot, nbins=nbins)
```

Comment:
@alejoe91 would it be possible to change the dropdown and have the plot work automatically, like this?

```suggestion
        widget.controls.column_dropdown.value = cols_to_plot
```

Reply:
Could you also add lines for each plot type, and that tests the error catching? I'd like the new widget to be fully covered by tests.

Reply:
No exceptions are thrown, but the dropdown is reset if combinations are not supported. Adding tests for those now